### PR TITLE
`Development`: Improve code quality of competency management

### PR DESCRIPTION
**Pull request by rstief**
# ONLY TESTABLE ON TS9, NEEDS IRIS.

#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I translated all newly inserted strings into English and German.

### Description
Improved code quality of competency management in multiple places:
- Extracted the competency taxonomy select to its own component to improve re-usability
- Removed function calls from multiple templates (indent of taxonomy select, translation of taxonomy tooltips, ...)
- Unified usage of "?" - Icon for missing competency taxonomies
- (Piggyback) Made header of competency recommendations clickable

### Steps for Testing
- 1 Instructor/Admin

1. Log in to Artemis
2. Go to [Raphael Test Course](https://artemis-test9.artemis.cit.tum.de/course-management/22/competency-management)
3. Click on `Create competency`
4. Verify that when entering "create"/"remember"/etc. in title/description the recommendation for a taxonomy gets displayed
5. Verify that taxonomy select works
6. Verify the competency gets created with the selected taxonomy.
7. Edit the competency you just created
8. See that the taxonomy gets displayed correctly and it gets saved correctly.
9. Click on `Generate Competency Recommendations`
10. Enter any course description, e.g. [introduction_to_programming.txt](https://github.com/ls1intum/Artemis/files/14139288/introduction_to_programming.txt)
11. Verify that the taxonomy select works here aswell for the competency reccommendations
12. Verif that competencies get created with the selected taxonomies

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [competency-form.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6063/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/competencies/competency-form/competency-form.component.ts.html) | 85.86% | ✅  |
| [competency-recommendation-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6063/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/competencies/generate-competencies/competency-recommendation-detail.component.ts.html) | 94.73% | ✅  |
| [taxonomy-select.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6063/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/competencies/taxonomy-select/taxonomy-select.component.ts.html) | 100% | ✅ |

### Screenshots
No UI changes.

### CODE REVIEW

**Review by reschandreas — DISMISSED** [review]
tested on TS9 during testing session, works as expected

**Review by laurenzfb — DISMISSED** [review]
retested on ts9, works fine

**Review by milljoniaer — APPROVED** [review]
code looks good 🚀

**Review by mateusmm01 — DISMISSED** [review]
Tested on ts9. Works as expected

File: src/main/webapp/app/course/competencies/competency-form/competency-form.component.ts
Comment (Strohgelaender) [inline src/main/webapp/app/course/competencies/competency-form/competency-form.component.ts:175]: This looks strange, can you explain this? Especially the `| undefined` part?

Comment (rstief) [reply src/main/webapp/app/course/competencies/competency-form/competency-form.component.ts:175]: undefined is an acceptable value, as a competency does not 100% need a competency.
The cast is used to create a form control of the type `Competency Taxonomy | undefined` with the formBuilder.

**Comment by MaximilianAnzinger** [comment]
@Strohgelaender do you have any further feedback?

**Review by milljoniaer — DISMISSED** [review]
Re-approve after merge conflicts

**Review by laurenzfb — APPROVED** [review]
reapprove

**Review by reschandreas — APPROVED** [review]
reapprove

File: src/main/webapp/app/entities/competency.model.ts
Comment (Strohgelaender) [inline src/main/webapp/app/entities/competency.model.ts:133]: This seems unnecessary, you could just build the tooltip via `'artemisApp.competency.taxonomies.' + taxonomy`.

**Review by milljoniaer — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]
code lgtm

**Review by laurenzfb — APPROVED** [review]

**Review by mateusmm01 — APPROVED** [review]
reapprove

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by bassner — APPROVED** [review]

**Pull request merged**
